src/ucp/core/ucp_worker.c

```diff
@@ -844,8 +844,9 @@ static int ucp_worker_iface_find_better(ucp_worker_h worker,
             /* 2. Has the same or better performance characteristics */
             (if_iter->attr.overhead <= wiface->attr.overhead) &&
             (ucp_tl_iface_bandwidth(ctx, &if_iter->attr.bandwidth) >= bw_cur) &&
-            (if_iter->attr.priority >= wiface->attr.priority) &&
-            (ucp_score_cmp(latency_iter, latency_cur) <= 0) &&
+            /* swap latencies in args list since less is better */
+            ucp_score_is_greater(latency_cur,  if_iter->attr.priority,
```

Comment:
this is not the same check, because it will discard interfaces which have the same latency and priority (even if they have better other characteristics).

Reply:
@brminich thanks for review!
@yosefe, this is the reason of failures. I'm going just to revert last commit, are you ok?

Reply:
`ucp_score_is_greater` is replaced with `ucp_score_prio_cmp`